src/python/pants/backend/python/target_types.py

```diff
@@ -587,16 +587,25 @@ class ModuleMappingField(DictStringToStringSequenceField):
         "A mapping of requirement names to a list of the modules they provide.\n\nFor example, "
         '`{"ansicolors": ["colors"]}`. Any unspecified requirements will use the requirement '
         'name as the default module, e.g. "Django" will default to `["django"]`.\n\nThis is '
-        "used for Pants to be able to infer dependencies in BUILD files."
+        "used to infer dependencies."
     )
     value: FrozenDict[str, Tuple[str, ...]]
 
+    @staticmethod
+    def normalize_project_name(s: str) -> str:
+        return s.lower().replace("-", "_")
```

Comment:
This is not a the right normalization: https://www.python.org/dev/peps/pep-0503/#normalized-names
You want something like: https://github.com/pypa/packaging/blob/20.9/packaging/utils.py#L39-L43

Reply:
Ah ha. Thanks! Now using `packaging.utils.canonicalize_name`